docs/notebooks/proton_radiography_particle_tracing_custom_source.ipynb

```diff
@@ -0,0 +1,524 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Proton Radiography: Synthetic Radiographs with Custom Source Profiles"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "An experimental reality of charged particle radiography experiments is that the source of the particles has finite size and often generates a non-uniform angular velocity distribution. For this reason, the`plasmapy.diagnostics.proton_radiography.SyntheticProtonRadiograph` particle tracing algorithm allows users to specify their own initial particle positions and velocities. This example will demonstrate how to use this functionality to create a more realistic synthetic radiograph that includes these source profile effects."
```

Comment:
```suggestion
    "[SyntheticProtronRadiography]: ../../api/plasmapy.diagnostics.proton_radiography.SyntheticProtonRadiograph.rst#syntheticprotonradiograph\n",
    "\n",
    "In the real world the particle source for charged particle radiography experiments is finite in size and often has a non-uniform angular velocity distribution. For this reason, the [SyntheticProtronRadiography] particle tracing algorithm allows users to specify their own initial particle positions and velocities. This example will demonstrate how to use this functionality."
```

Sphinx directives do not work in notebooks, so when referencing code documentation we have to used mardown's linking mechanisms.

Reply:
I find this pretty confusing, can you elaborate? There is no `/api` folder that I can see, nor a `SyntheticProtonRadiograph.rst`, so I assume these are files that are being created when the documentation is being compiled? How do I reference a method like `create_particles`?

Is there a way to get to this reference path from the documentation URL? They look similar...
```
https://docs.plasmapy.org/en/latest/api/plasmapy.diagnostics.proton_radiography.SyntheticProtonRadiograph.html#plasmapy.diagnostics.proton_radiography.SyntheticProtonRadiograph
```

Reply:
Everything in the `docs/api/` folder is auto generated by the `automodsumm` and `autosummary` directives.  That's why you don't see the files until you do a build of the documentation.  The naming scheme of the `api/` files is the same as what we use for the `api_static/` files, `plasmapy.mod.mod.Class`, `plasmapy.mod.mod.func`, etc.  The `#` at the end tells the build you're trying to reference a specific heading in that file.  To reference the whole page then you just exclude the `#` part.

![image](https://user-images.githubusercontent.com/29869348/110256360-62b26580-7f4d-11eb-8169-ae655503b4f7.png)

---
Here's the [`nbsphinx` documentation on this topic.](https://nbsphinx.readthedocs.io/en/0.8.2/markdown-cells.html#Links-to-Domain-Objects)

Reply:
I'll have to quote that comment into the doc guide...

Reply:
I've tried to do this, but I must still be doing something wrong? I have

```
[SyntheticProtronRadiography]: ../../api/plasmapy.diagnostics.proton_radiography.SyntheticProtonRadiograph.rst#plasmapy.diagnostics.proton_radiography.SyntheticProtonRadiograph

[SyntheticProtronRadiography] 
```

but CI raises this error

```
Warning, treated as error:
/home/runner/work/PlasmaPy/PlasmaPy/docs/notebooks/proton_radiography_particle_tracing_custom_source.ipynb:20:File not found: '../api/plasmapy.diagnostics.proton_radiography.SyntheticProtonRadiograph.rst#plasmapy.diagnostics.proton_radiography.SyntheticProtonRadiograph'
ERROR: InvocationError for command /home/runner/work/PlasmaPy/PlasmaPy/.tox/build_docs/bin/sphinx-build docs docs/_build/html -W -b html (exited with code 2)
```

Reply:
Your link is going back one directory too far, `../../` is sending you back to the repor root directory instead of `/docs/`.  Just delete one `../` and it should work.